src/main/groovy/edu/oregonstate/mist/catalogapitest/resources/InstructorResource.groovy

```diff
@@ -0,0 +1,210 @@
+    }
+
+    // /instructor -----------------------------------------------------------------------------------------------------
+
+    /**
+     * Responds to POST requests by creating and returning an instructor.
+     *
+     * @param newInstructor
+     * @return response containing the result or error message
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response postInstructor(@Valid Instructor newInstructor) {
+
+        Response returnResponse
+        URI createdURI
+
+        try {
+            instructorDAO.postInstructor(newInstructor.getFirstInitial(),
+                    newInstructor.getLastName(), newInstructor.getNumberOfCourses())
+
+            //TODO Add in the URI of newly created resource
+            createdURI = URI.create(uriInfo.getPath() + "/" + instructorDAO.getLatestCidNumber())
+            returnResponse = Response.created(createdURI).build()
+
+        } catch (UnableToExecuteStatementException e) {
+
+            String constraintError = e.getMessage()
+            ErrorPOJO returnError
+
+            if (constraintError.contains("INSTRUCTORS_UK_CID")) {
+
+                // CID number is not unique
+                returnError = new ErrorPOJO("CID is not unique", Response.Status.CONFLICT.getStatusCode())
+
+            } else {
+
+                // Some other error, should be logged
+                System.out.println(e.localizedMessage)
+                returnError = new ErrorPOJO("Unknown Error", Response.Status.INTERNAL_SERVER_ERROR.getStatusCode())
+            }
+
+            System.out.println(returnError.getErrorMessage())
+        }
+
+        return returnResponse
+    }
+
+    /**
+     * Responds to GET requests and retrieves all instructor objects
+     *
+     * @return list of all instructors, otherwise empty
+     */
+    @GET
+    @Path('/all')
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<Instructor> getByCid() {
+        return instructorDAO.allInstructors
+    }
+
+    // CID specific requests -------------------------------------------------------------------------------------------
+
+    /**
+     * Responds to GET requests and retrieves instructor with a specific cid.
+     *
+     * @param cid
+     * @return response containing the result or error message
+     */
+    @GET
+    @Path('{cid}')
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getByCid(@PathParam('cid') IntParam cid) {
+        Instructor instructors = instructorDAO.getByCid(cid.get())
+
+        Response returnResponse
+
+        if (instructors == null) {
+            ErrorPOJO returnError = new ErrorPOJO("Resource Not Found.", Response.Status.NOT_FOUND.getStatusCode())
+            returnResponse = Response.status(Response.Status.NOT_FOUND).entity(returnError).build()
+        } else {
+            returnResponse = Response.ok(instructors).build()
+        }
+
+        return returnResponse
```

Comment:
although it is just a matter of style, it would be interesting to know the cause of your error